File: poh20/javascript/8.js
#!/usr/bin/env node
B=[448,56,7,292,146,73,273,84]
A=parseInt((require('fs').readFileSync(0)+'').replaceAll(/./sg,e=>e=='\n'?'':+(e=='#')),2)
console.log(B.filter(b=>(A&b)==b).length)

File: poh20/java/8.java
//usr/bin/env java $0 $@;exit
import java.util.*;
interface Main{static void main(String[]z){
Scanner x=new Scanner(System.in).useDelimiter("\\Z");
int a=Integer.valueOf(x.next().replace("\n","").replace(".","0").replace("#","1"),2);
System.out.println(Arrays.asList(new Integer[]{448,56,7,292,146,73,273,84}).stream().filter(e->(a&e)==e).count());
}}
